src/indieweb_utils/posts/discovery.py

```diff
@@ -320,15 +320,17 @@ def get_post_type(h_entry: dict, custom_properties: List[Tuple[str, str]] = [])
     title = post.get("name")[0].strip().replace("\n", " ").replace("\r", " ")
 
     content = post.get("content")
+    content_to_validate = None
 
-    if content and content[0].get("text") and content[0].get("text")[0] != "":
-        content = BeautifulSoup(content[0].get("text"), "lxml").get_text()
+    if content:
+        if content[0].get("text") and content[0].get("text")[0] != "":
+            content_to_validate = BeautifulSoup(content[0].get("text"), "lxml").get_text()
 
-    if content and content[0].get("html") and content[0].get("html")[0] != "":
-        content = BeautifulSoup(content[0].get("html"), "lxml").get_text()
+        if content_to_validate is None and content[0].get("html") and content[0].get("html")[0] != "":
+            content_to_validate = BeautifulSoup(content[0].get("html"), "lxml").get_text()
```

Comment:
I think we could simplify this a bit... what do you think?

```python
# Default should be a list so we're never dealing with None
content = post.get("content", []) 
if content:
    # Default should be an empty string, so we're never dealing with None
    text = content[0].get("text", "")
    html = content[0].get("html", "")
    if html or text:
        # Prefer to validate against html than text version of the content
        content_text = BeautifulSoup(html or text, "lxml").get_text()
        if content_text and content_text.startswith(title):
            post_type = "article"
    return post_type

Reply:
I like it! I have just written a commit with your solution for your review. I unindented the `return` statement since `note` is set as the post type earlier in the function and this function isn't going to override that unless the conditions are met.